### PR TITLE
support https proxy

### PR DESCRIPTION
**Pull request by kawais**
support https proxy in options
const client = Binance( {
    apiKey: 'xxx',
    apiSecret: 'xxx',
    httpsProxy: 'http://127.0.0.1:1080'
} )

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ declare module 'binance-api-node' {
     httpFutures?: string
     wsBase?: string
     wsFutures?: string
+    httpsProxy?: string
```
Comment (balthazar) [inline index.d.ts:11]: can you call that `proxy`?

File: package.json [hunk 2]
```diff
@@ -16,6 +16,7 @@
     "lint": "eslint src"
   },
   "dependencies": {
+    "https-proxy-agent": "^5.0.0",
```
Comment (balthazar) [inline package.json:19]: this has been unmaintained for a year now, is there another solution?

File: src/http-client.js [hunk 3]
```diff
@@ -111,19 +113,23 @@ const checkParams = (name, payload, requires = []) => {
  * @param {object} headers
  * @returns {object} The api response
  */
-const publicCall = ({ endpoints }) => (path, data, method = 'GET', headers = {}) =>
-  sendResult(
+const publicCall = ({ httpsProxy, endpoints }) => (path, data, method = 'GET', headers = {}) => {
+  let opt = {
```
Comment (balthazar) [inline src/http-client.js:117]: `const` and call it `opts`

File: src/http-client.js [hunk 4]
```diff
@@ -111,19 +113,23 @@ const checkParams = (name, payload, requires = []) => {
...
+    method,
+    json: true,
+    headers,
+  }
```
Comment (balthazar) [inline src/http-client.js:121]: newline

File: src/http-client.js [hunk 5]
```diff
@@ -111,19 +113,23 @@ const checkParams = (name, payload, requires = []) => {
...
+  if (httpsProxy) opt.agent = new proxyAgent(httpsProxy)
```
Comment (balthazar) [inline src/http-client.js:122]: use brackets, even for single lines

File: src/http-client.js [hunk 6]
```diff
@@ -177,17 +183,17 @@ const privateCall = ({ apiKey, apiSecret, endpoints, getTime = defaultGetTime, p
       .digest('hex')
 
     const newData = noExtra ? data : { ...data, timestamp, signature }
-
+    let opt = {
+      method,
+      headers: { 'X-MBX-APIKEY': apiKey },
+      json: true,
+    }
+    if (httpsProxy) opt.agent = new proxyAgent(httpsProxy)
```
Comment (balthazar) [inline src/http-client.js:191]: avoid duplicating statements, this logic should be put in a function

File: src/open-websocket.js [hunk 7]
```diff
@@ -1,14 +1,15 @@
...
-export default url => {
+export default (url,proxyOpt) => {
   const rws = new ReconnectingWebSocket(url, [], {
     WebSocket: ws,
     connectionTimeout: 4e3,
     debug: false,
     maxReconnectionDelay: 10e3,
     maxRetries: Infinity,
     minReconnectionDelay: 4e3,
+    ...proxyOpt
```
Comment (balthazar) [inline src/open-websocket.js:12]: I dont see an option on reconnecting-websocket for a proxy, are you sure about this?

**Review by balthazar — CHANGES_REQUESTED** [review]

**Comment by kawais** [comment]
It's better to use https://github.com/gajus/global-agent.
so this pr is not necessary.

**Comment by balthazar** [comment]
Closing in favor of #513

**Pull request closed without merging**